src/wren/FrameBuffer.cpp

```diff
@@ -179,6 +179,27 @@ namespace wren {
     glstate::bindPixelPackBuffer(currentPixelPackBuffer);
   }
 
+  void FrameBuffer::copyContentsPart(size_t index, void *data, int src, int size) {
+    assert(index < mOutputDrawBuffers.size());
+    assert(mOutputDrawBuffers[index].mGlNamePbo);
+    assert(src + size < mWidth * mHeight);
+
+    const unsigned int currentPixelPackBuffer = glstate::boundPixelPackBuffer();
+    glstate::bindPixelPackBuffer(mOutputDrawBuffers[index].mGlNamePbo);
+
+    const Texture::GlFormatParams &params = drawBufferFormat(index);
+    const int sizeInBytes = params.mPixelSize * size;
+    const int srcInBytes = params.mPixelSize * src;
+
+    void *start = glMapBufferRange(GL_PIXEL_PACK_BUFFER, srcInBytes, sizeInBytes, GL_MAP_READ_BIT);
+    assert(start);
+    memcpy(data, start, sizeInBytes);
```

Comment:
`memcpy` of images are evil (performance disaster). We should avoid them. I believe we could use `glGetBufferSubData` instead of `glMapBufferRange` to avoid this `memcpy`.